GUIDELINES.md

```diff
@@ -0,0 +1,135 @@
+    * Functions starting with `then` should perform assertions (e.g. `thenIsLoggedIn()`, `thenIsBusy()` or `thenUserIsCreated(user: User)`)
+    * By starting at the test to define which functions are needed in the steps you are more focused and get a faster overview of that's needed instead of getting into details too early
+  * Now the test should already be human-readable for non-tech people (with just the technical implementation in the steps class missing)
+  * Remember that in the test class all data structures and function calls should be domain-specific and therefore be modelled around real-world domain objects and data visible to a user, as far as possible (if the test subject is of total technical nature this doesn't apply of course)
+* Now think about which dependencies (classes or interfaces) will be under test and add new steps classes
+  * You can add a steps classs per class under test (e.g. `AuthManagerSteps`)
+  * As stated above (just to repeat)
+    * the steps classes' job is to know "how this class is to be tested"
+    * these will contain the `given`, `when` and `then` functions
+    * and if reasonable you can split steps classes by responsibility rather than classes, as in generally it's considered a better approach to test against concepts under test, not concrete classes; but in practice this turns out to be very hard to achieve. Testing against interfaces instead of concrete classes would be a good compromise between ease of test implementation and concern separation.
+    * If the steps class is purely here for faking or mocking the real behavior of a class you should name it accordingly (e.g. `AuthManagerMockSteps` or `BackendGatewayFakeSteps`)
+    * To conclude, a steps class usually can be considered responsible for inetracting with or mocking/faking the behavior of a class or interface under test
+  * You also need to think of dependencies
+    * A dependency in sweetest can be considered as an object that is needed by another or multiple other objects and usually contains logic
+    * Dependencies in sweetest are considered singletons
+    * You can tell sweetest which dependencies there are in the test system to let it automatically create and inject dependencies into constructors whenever certain dependencies are needed
+    * So when you request a dependency of a specific type it's automatically constructed and further dependencies in the constructor are automatically initialized as well (that relieves you of the burden of putting together huge test setups when having a lot of dependencies, which is even more cumbersome when writing integration tests)
+    * In essence, sweetest has an own small dependency injection framework included
+  * A dependency can be either mocked or real
+    * By default in sweetest all dependencies are considered mocks
```

Comment:
Is this the case? But when I don't add it to the config, the tests don't run

Reply:
...and I think this is the correct behavior. Automatically assume mock can lead to hard debugging sessions.

Reply:
what i wanted to say here is, if something isn't specified as real it's automatically a mock, will update the wording...